### ISSUE TITLE
Add pre-existing HTML to gatsby blog template

### ISSUE THREAD

**Issue by ekeleshian** [gatsbyjs/gatsby#10594]
I'm trying to integrate pre-existing html into a blog website that originated from the Gatsby blog template.
I tried making a new directory inside the public folder and move the html file there.   I then typed my website's url slash the name of the html file, but the Not Found page appeared.   Just to double check that the page exists on my github pages, curling the route does give me the html file.  In the browser, though, this is what I get: 
![screenshot from 2018-12-20 16-34-42](https://user-images.githubusercontent.com/33647474/50318326-3944b680-0475-11e9-838d-0779c5f24beb.png)

Appreciate guidance on proper way to integrate existing html file into a gatsby project.

**Comment by sidharthachatterjee** [comment]
@ekeleshian Could you link to a minimal reproduction repo if possible?

**Comment by ekeleshian** [comment]
Here's the link to the repo:
 https://github.com/ekeleshian/ekeleshian.github.io/tree/developed
<https://github.com/ekeleshian/ekeleshian.github.io/tree/developed>
Just as a recap: I'm having a hard time fully integrating my pre-existing
code into my gatsby blog.  This code is located in public/ and are called
visualizations.html and new_vis.js.  When I type
ekeleshian.github.io/visualizations.html, it indeed directs me to the
custom made page, but when I click on a button, it won't call the function
that's supposed to be triggered in my js code.
Any ideas?

**Comment by DSchau** [comment]
This seems to be an issue with `gatsby-plugin-offline`, specifically because this page was added outside of the Gatsby lifecycle, the plugin isn't aware of that page.

Could you try removing `gatsby-plugin-offline` and check if this fixes your issue?

**Comment by ekeleshian** [comment]
Removing gatsby-plugin-offline unfortunately didn't fix the issue

**Comment by DSchau** [comment]
This error is _from_ `gatsby-plugin-offline`

![image](https://user-images.githubusercontent.com/3924690/50357245-e7ef0280-051a-11e9-8a02-acbfdd4ab6cb.png)

Perhaps you need to clear out the service worker? Additionally, you could investigate [`gatsby-plugin-remove-service-worker`](https://www.npmjs.com/package/gatsby-plugin-remove-serviceworker)

<img width="845" alt="screen shot 2018-12-21 at 12 21 37 pm" src="https://user-images.githubusercontent.com/3924690/50357273-fdfcc300-051a-11e9-834f-305ad279327d.png">

**Comment by ekeleshian** [comment]
Thanks so much. This indeed worked.

**Comment by ethan-cao** [comment]
> OK - so what's happening is that the `public` folder is being wiped. In general, you want to use the `static` folder for assets that will be directly copied to the `public` folder.
> 
> e.g. [this guide is helpful](https://www.gatsbyjs.org/docs/adding-images-fonts-files/#using-the-static-folder)
> 
> So to fix this, you'll want to copy `public/*` (e.g. those two files) into `static`. The build process will copy these files for you into `public` and then they should be routable in the resulting build.
> 
> Hope this helps!
> 
> <img alt="screen shot 2018-12-21 at 12 29 14 pm" width="913" src="https://user-images.githubusercontent.com/3924690/50357588-099cb980-051c-11e9-8405-8d8253018cd4.png">

I add a existing html file test.html to gatsby_project/static, and run gatsby develop. After everything is up and running, I tried to visit http://localhost:8000/test.html and got 404 page, in which it says Gatsby.js development 404 page, There's not a page yet at /test.html.

Did I miss something ? 

Appreciate any help.

**Comment by TimFletcher** [comment]
Yeah, same issue. Seems to work if you do a `gatsby build && gatsby serve` though.
